### PR TITLE
Improve sensor interval

### PR DESCRIPTION
**Pull request by chrisruk**

### CODE REVIEW

File: buildhat/serinterface.py [hunk 1]
```diff
@@ -138,6 +140,7 @@ def __init__(self, firmware, signature, version, device="/dev/serial0", debug=Fa
                 else:
                     self.write(b"version\r")
 
+        self.state = HatState.FIRMWARE
```
Comment (loiswells97) [inline buildhat/serinterface.py:143]: From the commit message, does this need to be removed?

File: test/color.py [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+        end = time.time()
+        diff = abs((end - start) - expecteddur)
+        self.assertLess(diff, 0.25)
+
+        start = time.time()
+        for _ in range(count):
+            color.get_color_rgbi()
+        end = time.time()
+        diff = abs((end - start) - expecteddur)
+        self.assertLess(diff, 0.25)
```
Comment (loiswells97) [inline test/color.py:31]: Would these make more sense as separate tests with a before block for the setup?

File: test/color.py [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+class TestColor(unittest.TestCase):
+    """Test color sensor functions"""
+
+    def test_color(self):
+        """Test color sensor interval"""
+        color = ColorSensor('A')
+        color.avg_reads = 1
+        color.interval = 10
+        count = 1000
+        expecteddur = count * color.interval * 1e-3
```
Comment (loiswells97) [inline test/color.py:17]: Might be more readable in snake_case

**Review by loiswells97 — COMMENTED** [review]
Just a few small comments

File: buildhat/serinterface.py (hunk 1 above)
Comment (chrisruk) [reply buildhat/serinterface.py]: Good point, just dropped

File: test/color.py (hunk 3 above)
Comment (chrisruk) [reply test/color.py:17]: You mean expeced_dur rather than expecteddur?

Comment (loiswells97) [reply test/color.py:17]: Yeah

File: test/color.py (hunk 2 above)
Comment (chrisruk) [reply test/color.py:31]: I can see what you mean but they're both just testing the setting of the interval works correctly.  Would it be ok if I left as-is and called the function something like ```test_color_interval```

Comment (loiswells97) [reply test/color.py:31]: Okay then, let's go with that

Comment (chrisruk) [reply test/color.py:31]: Just done

**Review by loiswells97 — APPROVED** [review]
Looks good now 👍

**Pull request merged**